core/domain/user_services.py

```diff
@@ -188,6 +191,13 @@ def validate(self):
                 self.gae_id
             )
 
+        if (self.pin is not None and
```

Comment:
is there any (max-min) length limitation for this value? If yes, do we need to validate that?

Reply:
@DubeySandeep Yes that does seem like a valid check to do. Thanks!

@BenHenning I wanted to know your views on this since I am a bit unsure - I think the max length allowed for PIN is 5 and min length is 3 (if pin is not None), right? (Saying this by using the android app) 
Also two ways for storing PIN are:
1) We either store the pin
2) Or store a hash of the PIN
As we are treating PIN merely as a storage space for backup on the backend side, it totally depends on what we are receiving from the android - (1) or (2).
If it sounds okay, can I proceed with assumption that min=3,max=5 length of the pin is only possible for now, and it can be modified later on? Sounds okay? @BenHenning

Reply:
Max is 5 for device admins and 3 for all other profile types. The PIN is coming from Android, and we haven't yet finalized whether it's going to be synced across devices (but I suspect it will need to be otherwise login will be awkward when you set a new primary device for an account). Based on that:
- Validation 3-5 characters sounds good
- Keeping it plain text for now sounds good

We very may likely change these in the future, but these checks seem correct given current information.

Reply:
Just to confirm there are just 3 possibilities:
1) Pin is None
2) And if the PIN is not None, then for full users `PIN` is of length **5**
3) And for profile users `PIN` is of length **3**

Right @BenHenning ?

Reply:
Yes this is OK, make sure to validate the charset too.